cp23/cubes/cubes/executables/solution.cpp

```diff
@@ -0,0 +1,49 @@
+#include <bits/stdc++.h>
+
+using namespace std;
+
+typedef long long ll;
+
+typedef struct
+{
+    ll x;
+    ll y;
+    ll z;
+} Cube;
+
+vector<Cube> cubes;
+
+static bool adjacent(Cube a, Cube b)
+{
+    return (abs(a.x - b.x) + abs(a.y - b.y) + abs(a.z - b.z)) == 1;
+}
+
+static ll getSurfaceArea()
+{
+    ll total = 6 * cubes.size();
+    for (int i = 0; i < cubes.size(); i++)
+        for (int j = i + 1; j < cubes.size(); j++)
```

Comment:
@AbdullatifGhajar ich glaube man kann das auch lin linearer Zeit schaffen, wenn man ein Set z.b. nutzt und als key entweder string von `x + " " + y` nutzt, oder evtl. die beiden in einen 64 bit int zusammenfügt?
z.b. `x << 32 + y`

Reply:
Ja ein guter Ansatz. Du brauchst das natürlich für `x + " " + y`, `y + " " + z` und `z + " " + x`. Bei Sets hast du ein Log-Faktor, also O(n log n)